tests/test_pages.py

```diff
@@ -430,3 +436,151 @@ def nextPage(self):
         assert request.metadata is not None
         assert request.metadata.probability == 1.0
     assert type(page.nextPage) == Request
+
+
+def test_price_selector():
+    class CustomProductPage(ProductPage):
+        @field
+        def price(self):
+            return self.css("div::text")
+
+    html = b"""
+    <!DOCTYPE html>
+    <html>
+        <body>
+            <div>$13.2</div>
+        </body>
+    </html>
+    """
+    url = RequestUrl("https://example.com")
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.price == "13.2"
+    assert page.currency is None
+    assert page.currencyRaw == "$"
+
+    # access currency fields before the price field
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.currency is None
+    assert page.currencyRaw == "$"
+    assert page.price == "13.2"
+
+
+def test_price_explicit():
+    class CustomProductPage(ProductPage):
+        @field
+        def price(self):
+            return self.css("div::text").get()
+
+    html = b"""
+    <!DOCTYPE html>
+    <html>
+        <body>
+            <div>$13.2</div>
+        </body>
+    </html>
+    """
+    url = RequestUrl("https://example.com")
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.price == "$13.2"
+    assert page.currency is None
+    assert page.currencyRaw is None
+
+    # access currency fields before the price field
+    page = CustomProductPage(response=HttpResponse(url=url, body=html))
+    assert page.currency is None
+    assert page.currencyRaw is None
+    assert page.price == "$13.2"
+
+
+def test_currency_hardcoded():
+    class CustomProductPage(ProductPage):
+        CURRENCY = "USD"
```

Comment:
Should we have a basic test, which checks CURRENCY in isolation? You don't need to define the price field for it to work, right?